main.rb

```diff
@@ -0,0 +1,142 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?
+    return (my_all? { |a| !a.nil? }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
+
+  def my_any?
+    return (my_any? { |x| !x.nil? }) unless block_given?
+
+    my_each do |value|
+      return true if yield value
+    end
+    false
+  end
+
+  def my_none?
+    return (my_none? { |x| !x.nil? }) unless block_given?
+
+    my_each do |val|
+      return false if yield val
+    end
+    true
+  end
+
+  def my_count(nval = nil)
+    cont = 0
+    if !nval.nil?
+      my_each do |value|
+        cont += 1 if nval == value
+      end
+    else
+      my_each do
+        cont += 1
+      end
+    end
+    cont
+  end
```

Comment:
4. `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. Counts the number of elements yielding a true value if a block is given
```ruby
block = proc { |num| num < (5 + 15) / 2 }
[1, 2 ,3, 5, 18, 19].my_count(&block) #should return 4
```